Feat: Card Frame 컴포넌트

i-nooo-2 opened the pull request:
## 🏷️ 이슈 번호

- close #14 

## 🧱 작업 사항
- **Card Frame Component 구현**
	- Card Frame, Content Wrapper로 구성
	- 입력받은 type에 따라 Content Wrappe의 구성이 바뀌도록 함.
	- 또한 type에 따라 props를 강제했기 때문에 정해진 props가 아닌 경우 컴파일 에러가 발생함.
	
![image](https://github.com/user-attachments/assets/44a8709e-2579-401c-9325-e9ecd5911bad)

- **format 유틸 함수**
	- formatNumberWithCommas : 숫자 인수를 받고, 3번째 자릿수마다 쉼표(,)를 찍어 문자열로 반환	
	- formatDate : (YYYY.M.D) 형식의 날짜 문자열을 반환
	- formatTimeRange : ISO 8601 형식 날짜 문자열 인수 2개를 받아 시간 범위를 문자열로 반환 (HH:mm-HH:mm)
- **tailwindCSS prettier 및 eslint의 규칙을 수정하였습니다.**
	- class 순서 자동 정렬 / 중복 클래스 제거 /  합칠 수 있는 클래스 존재 시 자동으로 합치도록
	- 등의 규칙이 적용되었습니다.

## 📸 결과물
![Animationasd](https://github.com/user-attachments/assets/dabd7725-3702-4296-9910-6e2822713694)

## 💬 공유 포인트 및 논의 사항
- 해당 컴포넌트를 이용한 개발 시 **추가적으로 수정되는 것을 염두에 두고** 만들었습니다.
	- 때문에 data interface, type이 api 응답 형식과 정확히 일치하지 않습니다.
	- 자유롭게 수정해서 이용해주세요 :)
- 예시 이미지에서 꾸겨지는 부분(?)은 의도한 부분입니다.
	- 해당 width는 모바일 표준 width에도 한참 못 미치는 width기 때문에 따로 작업하지 않았습니다.
- 앞서 제작된 공용 컴포넌트의 적용이 필요합니다.
	- **Button, Rating component의 적용이 필요합니다.**

## 수정 사항
- [x] breakpoint 추가 및 해당 컴포넌트 내 breakpoint 수정

i-nooo-2 committed:
Feat: Card Frame 컴포넌트 하드 코딩

i-nooo-2 committed:
Feat: 콘텐츠 type에 따라 contents props 강제

i-nooo-2 committed:
Design: 반응형 Screens 수정

- max width 기준으로 반영되도록 하였습니다.
- mobile 조건에 부합할 시, tablet보다 우선적으로 적용되도록 max-width 미디어 쿼리의 정의된 순서를 수정하였습니다.

i-nooo-2 committed:
Feat: card frame 스타일링 및 반응형 대응 / contentsWrapper 분리

i-nooo-2 committed:
Chore: tailwindcss eslint, prettier plugin 적용

i-nooo-2 committed:
Feat: ContentsFrame 스타일링 및 반응형 대응

- 체험 영문명 변경 : Experience -> Activities

i-nooo-2 committed:
Feat: 날짜 format 변환 유틸 추가

- formatDate : (YYYY.M.D) 형식의 날짜 문자열을 반환
- formatTimeRange : ISO 8601 형식 날짜 문자열 인수 2개를 받아 시간 범위를 문자열로 반환 (HH:mm-HH:mm)

i-nooo-2 committed:
Refactor: 코드 순서 및 cn 적용 순서 변경

eunbinnie commented on app/_components/card-frame/index.tsx:
```diff
@@ -0,0 +1,27 @@
+  return (
+    <div className="group flex h-[204px] w-full cursor-pointer overflow-hidden rounded-3xl bg-white shadow-[0_4px_16px_0_rgba(17,34,17,0.05)] tablet:h-[156px] mobile:h-[128px]">
+      <div className="relative size-full max-h-[204px] max-w-[204px] overflow-hidden tablet:max-w-[156px] mobile:max-w-[128px]">
+        <Image
```
Image 컴포넌트에서 경고가 조금 있어서 수정하면 좋을 것 같습니다! 이렇게 테스트해보니까 경고가 사라졌습니다
```ts
<Image
  src={img}
  alt={`${contents.title} img`}
  fill
  sizes="max-width:100%"
  style={{ objectFit: 'cover' }}
  priority
  className="transform transition-transform duration-300 ease-in-out group-hover:scale-125"
/>
```

eunbinnie commented on app/_components/card-frame/_components/card-contents-wrapper/index.tsx:
```diff
@@ -0,0 +1,111 @@
+      <p className={cn(statusVariants({ status }))}>{status}</p>
+      <p className={titleStyle}>{contents.title}</p>
+      <div className="mb-[17px] flex items-center gap-2 whitespace-nowrap text-lg/6 font-normal tablet:mb-[10px] tablet:text-sm/6 mobile:mb-0 mobile:text-xs/6">
+        <p>{formatDate(contents.period)}</p> · <p>{formatTimeRange(contents.time1, contents.time2)}</p> · <p>{contents.headCount}명</p>
```
데이터가 YYYY-MM-DD , HH:MM 형태의 string으로 와서 formatDate, formatTimeRange 사용이 어려울 것 같습니다..! 만들어주셨는데..🥹

i-nooo-2 replied:
아아앗... .. 이건 좀 아쉬움이 남네요 ㅋㅋㅠㅠㅠㅠ

eunbinnie replied:
ㅠㅠ 다른 페이지에서 쓸 일이 있다면 좋겠군여 ..

eunbinnie reviewed (approved):
감사합니다! 만들어주신 컴포넌트 잘 사용하겠습니다 ❤️‍🔥 예약 내역 카드 프레임의 경우, 작업하면서 수정을 조금 할 수도 있을 것 같습니다! 궁금한 점은 여쭤보면서 작업하겠습니다 고생많으셨습니다!

edhcoding reviewed (approved):
고생하셨습니다!

yubugi reviewed (approved):
설명까지 자세히 적어주셨네요! 고생하셨습니다!

i-nooo-2 committed:
Design: breakpoint 추가 및 기존 코드에 적용

- under-* 추가 : max 의미

i-nooo-2 committed:
Fix: Thumbnail Image 태그 속성 수정

- sizes 속성 추가
- layout 속성 제거 및 fill 속성 사용 (Next.js 13 이상에서는 layout 속성이 더 이상 사용되지 않습니다.)
- objectFit="cover" 대신 style 속성을 사용하여 objectFit 설정 (공식에서 권장되는 방식으로 변경)

i-nooo-2 committed:
Merge branch 'develop' into feat/#14-card-frame-component
